File: コードレビュー/05.マスター/プログラム作成問題：マスター編/20200302_138666938/src/namebattler/Thunder.java
package namebattler;

public class Thunder extends Magic{
	// =======================
	// フィールド変数
	// =======================
	//ダメージ量
	private int damage;

	// =======================
	// コンストラクタ
	// =======================
	//コンストラクタ
	public Thunder() {
		super();
		//名前をセット
		this.name = "サンダー";
		//使用MPをセット
		this.cost = 20;
	}

	// =======================
	// protected メソッド
	// =======================
	//個別効果
	@Override
	protected void Magieffect(Player target) {
		//ダメージ量を設定(10～30)
		this.damage = this.ran.nextInt(21)+10;
		//ターゲットにダメージ
		System.out.print(target.GetName() + "に" + this.damage + "のダメージ！");
		target.Damage(this.damage);
	}
}

File: コードレビュー/07.Unityアプリ/20200701_USSBB1M39/Assets/Script/Thunder.cs
using System;
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class Thunder : Magic
{

	// =======================
	// フィールド変数
	// =======================
	// 魔法の名前
	protected string name = "サンダー";
	// 消費MP
	protected int usemp = 20;

	// =======================
	// コンストラクタ
	// =======================
	public Thunder()
	{

	}

	// =======================
	// Getter / Setter
	// =======================
	// 魔法のMPの取得
	public string GetName()
	{
		return this.name;
	}

	public  int GetUseMP()
	{
		return this.usemp;
	}

	// =======================
	// protected メソッド
	// =======================

	// =======================
	// private メソッド
	// =======================

	// =======================
	// public メソッド
	// =======================
	/**
	 * 対称プレーヤーに電撃を加える
	 * @param activePlayer {@inheritDoc}
	 * @param passivePlayer {@inheritDoc}
	 */
	public override void effect(Player activePlayer, Player passivePlayer)
	{

		// MPが足りている場合の処理
		int damage = UnityEngine.Random.Range(10, 30);

		activePlayer.UseMP(this.usemp);

		LogText.AddLog(string.Format("{0} の {1}！", activePlayer.GetName(), this.name));
		LogText.AddLog($"{passivePlayer.GetName()} に {damage} のダメージ！");

		passivePlayer.Damage(damage);
	}
}